lib/datasource/docker/index.ts

```diff
@@ -535,3 +540,22 @@ export async function getPkgReleases({
   }
   return ret;
 }
+
+function getECRAuthToken(region: string, opts: hostRules.HostRule) {
+  const config = { region, accessKeyId: undefined, secretAccessKey: undefined };
+  if (opts.username && opts.password) {
```

Comment:
What happens in the "else" situation here? Isn't this guaranteed to fail if there's only a region? Or is it possible to have ECR without username/password required?

Reply:
> Or is it possible to have ECR without username/password required

Yup, certainly is - it would leverage credentials in the chain. Check out the [AWS SDK Docs](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials-node.html). A typical use case could be (ie, my use case) is renovate running on a EC2 instance with a IAM role associated that explicitly allows access to private ECR repos